### ISSUE TITLE
New plugin DSL doesn't support generating Dynamic Libraries

### ISSUE THREAD

**Issue by luca992** [JetBrains/kotlin-native#1877]
According to [OutputKind](https://github.com/JetBrains/kotlin-native/blob/master/tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/OutputKind.kt) there are only three output types supported (EXECUTABLE, KLIBRARY, FRAMEWORK) with the new plugin: 

    apply plugin: 'kotlin-platform-native'

I would just use the konan plugin to generate it. But I need support for kotlinx.coroutines, and I do not think there is a way to use coroutines with the konan plugin?

So, is there any way to generate a dynamic library with the new plugin?

**Comment by ilmat192** [comment]
> I do not think there is a way to use coroutines with the konan plugin?

You're right. Theoretically it's possible to set the configuration attributes manually but it's too hacky.

> So, is there any way to generate a dynamic library with the new plugin?

Dynamic libraries are not supported at the moment but they are in short-term plans

**Comment by ilmat192** [comment]
Actually it's possible to hack the new plugin to generate a dynamic library while such a function isn't supported:
```
sourceSets.main.component {
    outputKinds = [ KLIBRARY ]
    extraOpts '-p', 'dynamic', '-module_name', baseName.get()
}
```

**Comment by luca992** [comment]
@ilmat192 good to know, I'll give it a try

**Comment by olonho** [comment]
Please report if it fixes your issue.

**Comment by luca992** [comment]
@olonho it works!

This is probably a separate issue. But, in the generated c header file, classes from other modules are not available.

For example here is my build.gradle generating the dynamic library

    apply plugin: 'kotlin-platform-native'



    dependencies {
        implementation project(':data')
        implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core-native:0.24.0'
    }

    sourceSets {
         main {
            component {
               outputKinds = [ KLIBRARY ]
               extraOpts '-p', 'dynamic', '-module_name', baseName.get()
            }
        }
    }

I would like classes from `project(':data')` to be included inside the dynamic library header. Is there anyway to specify that?

**Comment by olonho** [comment]
Shall be fixed with aee041fc7aba0dec2eb0053c8a90e4e5f2bd032c. No reexport is how this feature is designed, only explicitly mentioned in source code functions and classes are exported.
